src/credentials/activation.ts

```diff
@@ -0,0 +1,30 @@
+/*!
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { AwsContext } from '../shared/awsContext'
+import { CredentialsProfileMru } from '../shared/credentials/credentialsProfileMru'
+
+export interface CredentialsInitializeParameters {
+    extensionContext: vscode.ExtensionContext
+    awsContext: AwsContext
+}
+
+export async function initialize(parameters: CredentialsInitializeParameters): Promise<void> {
+    updateMruWhenAwsContextChanges(parameters.awsContext, parameters.extensionContext)
+}
+
+function updateMruWhenAwsContextChanges(awsContext: AwsContext, extensionContext: vscode.ExtensionContext) {
+    extensionContext.subscriptions.push(
+        awsContext.onDidChangeContext(async awsContextChangedEvent => {
+            if (!awsContextChangedEvent.profileName) {
+                return
+            }
+
+            const mru = new CredentialsProfileMru(extensionContext)
```

Comment:
If we're sticking with class-based, we should move this const out of this event so we don't create a new one every time we see the event. Maybe move it to line 20, outside the subscription registration?